### DISCUSSION TITLE
Labels for small paths

### DISCUSSION THREAD

**Discussion by pstorch** [mapsforge/vtm#1104] [Q&A] [answered]
Hi, I've migrated one of my apps from Mapsforge to VTM. It works great so far. One "small" issue is labels of small paths are not shown in VTM.

Here you see the VTM rendered map with no path label:
![Screenshot_20240224_095002_OSM Dashboard (Nightly)](https://github.com/mapsforge/vtm/assets/5598124/9a5ef5e5-d25d-4265-a13c-524d412d81c0)

Here is the mapsforge rendered map with the path label:
![Screenshot_20240225_115832_Railway station photos](https://github.com/mapsforge/vtm/assets/5598124/2bcbfa84-e2a0-4e19-925b-e0bf2e2a1142)

How do I enable the drawing of path labels in VTM?
I use offline mapsforge maps with the default style.

**Comment by devemux86** [comment]
Have you checked the related tag in the VTM theme?
Does the name appear with other themes to compare?

**Reply by pstorch** [reply]
I've tried all themes in VtmThemes, but none show the label. Where do I have to look for the tag?
The example map is the hessen.map from here: http://download.mapsforge.org/maps/v5/europe/germany/

**Reply by devemux86** [reply]
You can find the tags of a path (or any map element) in openstreetmap.org.
Then search the themes to see how these tags are rendered.

**Reply by pstorch** [reply]
![image](https://github.com/mapsforge/vtm/assets/5598124/d1ffa983-7fab-496b-9750-ddb011a43ba0)

`highway` is `path`. Is that the tag I have to look for?

Is it this element in the vtm-themes file?: https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/default.xml#L139

**Reply by devemux86** [reply]
It probably doesn't have a text rule like the major highway tags.

**Reply by devemux86** [reply]
Search for `<text use="road" />` in the themes.
It should exist for roads and try something similar for paths.

**Reply by pstorch** [reply]
Thanks, I think I found it.

**Reply by pstorch** [reply, marked as answer]
If others come here with a similar issue, here is what I did:

I copied this file https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/default.xml into my own apps `src/main/assets` directory as `vtmstyle.xml`.

Then I added `<text use="road" />` to these elements:
![image](https://github.com/mapsforge/vtm/assets/5598124/7dcf291c-a5e7-43aa-8772-8634ec1ac3f9)

At last I changed this line in my code:

```java
        renderTheme = map.setTheme(VtmThemes.DEFAULT);
```

to this:

```java
        renderTheme = map.setTheme(new StreamRenderTheme("", AssetAdapter.readFileAsStream("vtm/vtmstyle.xml")));

```

**Comment by devemux86** [comment]
Also the 2nd image is an online map?
Not a Mapsforge offline map.

**Reply by pstorch** [reply]
Sorry, my bad. That was the wrong screenshot. I've corrected the original post.
